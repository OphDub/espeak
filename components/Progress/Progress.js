import React, { Component } from 'react';
import {
  View,
  StyleSheet,
  Text
} from 'react-native';
import PropTypes from 'prop-types';

export default class Progress extends Component<Props> {
  styleBar = (currentStatus) => {
    let percentComplete = (currentStatus/this.props.wordTotal) * 100;
    return {
      width: `${percentComplete}%`,
      height: 16,
      backgroundColor: '#F5E663',
      borderRadius: 2,
      borderColor: '#F5E663',
      borderLeftWidth: 2
    };
  }

  render() {
    const { category } = this.props.stack;
 
    return (
      <View style={styles.container}>
        <View style={styles.pointsWrapper}>
          <View style={styles.progressWrapper}>
            <View style={this.styleBar(this.props.currentCardId)}></View>
          </View>
          <Text style={styles.points}>
            {this.props.userPoints} pts
          </Text>
        </View>
        <Text style={styles.text}>
          {category}: {this.styleBar(this.props.currentCardId).width}
        </Text>
      </View>
    );
  }
}

Progress.propTypes = {
  stack: PropTypes.object,
  currentCardId: PropTypes.number,
  userPoints: PropTypes.number,
  wordTotal: PropTypes.number
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    justifyContent: 'center',
    position: 'absolute',
    top: 0,
    height: 63,
    width: '100%',
    alignItems: 'center',
    overflow: 'hidden',
    backgroundColor: '#B47AEA'
  },
  progressWrapper: {
    width: 260,
    height: 20,
    borderWidth: 2,
    borderRadius: 5,
    marginTop: 15,
    borderColor: '#F5E663'
  },
  pointsWrapper: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  text: {
    alignSelf: 'flex-start',
    marginTop: 5,
    marginLeft: 60,
    color: '#FFF'
  },
  points: {
    marginLeft: 15,
    fontSize: 20,
    color: '#FFF',
    alignSelf: 'flex-end'
  }
});