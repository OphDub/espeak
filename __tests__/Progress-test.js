import React from 'react';
import { shallow, configure } from 'enzyme';
import Progress from '../components/Progress/Progress';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const mockProp = {category: 'basics 1'}
  const wrap = renderer.create(<Progress stack={mockProp}/>).toJSON();
  expect(wrap).toMatchSnapshot();
})

it('it renders when a new stack is passed in as props', () => {
  const mockProp = { category: 'basics 1' }
  const mockPropTwo = { category: 'verbs' }
  const wrapper = shallow(
    <Progress stack={mockProp} />
  );
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ stack: mockPropTwo });
  expect(wrapper).toMatchSnapshot();
});
