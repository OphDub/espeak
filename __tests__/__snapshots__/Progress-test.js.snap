// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders when a new stack is passed in as props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Progress
    stack={
      Object {
        "category": "basics 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F5E663",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 20,
                "marginTop": 15,
                "width": 260,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F5E663",
                  "borderColor": "#F5E663",
                  "borderLeftWidth": 2,
                  "borderRadius": 2,
                  "height": 16,
                  "width": "NaN%",
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#FFF",
                "fontSize": 20,
                "marginLeft": 15,
              }
            }
          >
             pts
          </Text>
        </View>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "#FFF",
              "marginLeft": 60,
              "marginTop": 5,
            }
          }
        >
          basics 1
          : 
          NaN%
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#B47AEA",
        "display": "flex",
        "height": 63,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />
            </View>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                }
              }
            >
               pts
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />,
              "style": Object {
                "borderColor": "#F5E663",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 20,
                "marginTop": 15,
                "width": 260,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "backgroundColor": "#F5E663",
                  "borderColor": "#F5E663",
                  "borderLeftWidth": 2,
                  "borderRadius": 2,
                  "height": 16,
                  "width": "NaN%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                undefined,
                " pts",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "flex-end",
                "color": "#FFF",
                "fontSize": 20,
                "marginLeft": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " pts",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            "basics 1",
            ": ",
            "NaN%",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "flex-start",
            "color": "#FFF",
            "marginLeft": 60,
            "marginTop": 5,
          },
        },
        "ref": null,
        "rendered": Array [
          "basics 1",
          ": ",
          "NaN%",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                }
              }
            >
               pts
            </Text>
          </View>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "#FFF",
                "marginLeft": 60,
                "marginTop": 5,
              }
            }
          >
            basics 1
            : 
            NaN%
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#B47AEA",
          "display": "flex",
          "height": 63,
          "justifyContent": "center",
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "borderColor": "#F5E663",
                    "borderRadius": 5,
                    "borderWidth": 2,
                    "height": 20,
                    "marginTop": 15,
                    "width": 260,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#F5E663",
                      "borderColor": "#F5E663",
                      "borderLeftWidth": 2,
                      "borderRadius": 2,
                      "height": 16,
                      "width": "NaN%",
                    }
                  }
                />
              </View>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "#FFF",
                    "fontSize": 20,
                    "marginLeft": 15,
                  }
                }
              >
                 pts
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <View
                  style={
                    Object {
                      "backgroundColor": "#F5E663",
                      "borderColor": "#F5E663",
                      "borderLeftWidth": 2,
                      "borderRadius": 2,
                      "height": 16,
                      "width": "NaN%",
                    }
                  }
                />,
                "style": Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "style": Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  undefined,
                  " pts",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " pts",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "basics 1",
              ": ",
              "NaN%",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "flex-start",
              "color": "#FFF",
              "marginLeft": 60,
              "marginTop": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            "basics 1",
            ": ",
            "NaN%",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it renders when a new stack is passed in as props 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Progress
    stack={
      Object {
        "category": "verbs",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#F5E663",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 20,
                "marginTop": 15,
                "width": 260,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F5E663",
                  "borderColor": "#F5E663",
                  "borderLeftWidth": 2,
                  "borderRadius": 2,
                  "height": 16,
                  "width": "NaN%",
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#FFF",
                "fontSize": 20,
                "marginLeft": 15,
              }
            }
          >
             pts
          </Text>
        </View>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "#FFF",
              "marginLeft": 60,
              "marginTop": 5,
            }
          }
        >
          verbs
          : 
          NaN%
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#B47AEA",
        "display": "flex",
        "height": 63,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />
            </View>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                }
              }
            >
               pts
            </Text>,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />,
              "style": Object {
                "borderColor": "#F5E663",
                "borderRadius": 5,
                "borderWidth": 2,
                "height": 20,
                "marginTop": 15,
                "width": 260,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "backgroundColor": "#F5E663",
                  "borderColor": "#F5E663",
                  "borderLeftWidth": 2,
                  "borderRadius": 2,
                  "height": 16,
                  "width": "NaN%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                undefined,
                " pts",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "flex-end",
                "color": "#FFF",
                "fontSize": 20,
                "marginLeft": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " pts",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            "verbs",
            ": ",
            "NaN%",
          ],
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "flex-start",
            "color": "#FFF",
            "marginLeft": 60,
            "marginTop": 5,
          },
        },
        "ref": null,
        "rendered": Array [
          "verbs",
          ": ",
          "NaN%",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  }
                }
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                }
              }
            >
               pts
            </Text>
          </View>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "#FFF",
                "marginLeft": 60,
                "marginTop": 5,
              }
            }
          >
            verbs
            : 
            NaN%
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#B47AEA",
          "display": "flex",
          "height": 63,
          "justifyContent": "center",
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "borderColor": "#F5E663",
                    "borderRadius": 5,
                    "borderWidth": 2,
                    "height": 20,
                    "marginTop": 15,
                    "width": 260,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#F5E663",
                      "borderColor": "#F5E663",
                      "borderLeftWidth": 2,
                      "borderRadius": 2,
                      "height": 16,
                      "width": "NaN%",
                    }
                  }
                />
              </View>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "#FFF",
                    "fontSize": 20,
                    "marginLeft": 15,
                  }
                }
              >
                 pts
              </Text>,
            ],
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <View
                  style={
                    Object {
                      "backgroundColor": "#F5E663",
                      "borderColor": "#F5E663",
                      "borderLeftWidth": 2,
                      "borderRadius": 2,
                      "height": 16,
                      "width": "NaN%",
                    }
                  }
                />,
                "style": Object {
                  "borderColor": "#F5E663",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 20,
                  "marginTop": 15,
                  "width": 260,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "style": Object {
                    "backgroundColor": "#F5E663",
                    "borderColor": "#F5E663",
                    "borderLeftWidth": 2,
                    "borderRadius": 2,
                    "height": 16,
                    "width": "NaN%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  undefined,
                  " pts",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 20,
                  "marginLeft": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " pts",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "verbs",
              ": ",
              "NaN%",
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "flex-start",
              "color": "#FFF",
              "marginLeft": 60,
              "marginTop": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            "verbs",
            ": ",
            "NaN%",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#B47AEA",
      "display": "flex",
      "height": 63,
      "justifyContent": "center",
      "overflow": "hidden",
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#F5E663",
          "borderRadius": 5,
          "borderWidth": 2,
          "height": 20,
          "marginTop": 15,
          "width": 260,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F5E663",
            "borderColor": "#F5E663",
            "borderLeftWidth": 2,
            "borderRadius": 2,
            "height": 16,
            "width": "NaN%",
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#FFF",
          "fontSize": 20,
          "marginLeft": 15,
        }
      }
    >
       pts
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#FFF",
        "marginLeft": 60,
        "marginTop": 5,
      }
    }
  >
    basics 1
    : 
    NaN%
  </Text>
</View>
`;
